legacy_promotions/db/migrate/20160101010001_solidus_one_four_promotions.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+class SolidusOneFourPromotions < ActiveRecord::Migration[5.0]
+  def up
+    # This migration is just a compressed version of all the previous
+    # migrations for spree_core. Do not run it if one of the core tables
+    # already exists. Assume the best.
+    return if table_exists?(:spree_promotions)
+
+    create_table "spree_orders_promotions", force: :cascade do |t|
```

Comment:
need `if_not_exists: true` checks on all tables

Reply:
Added a commit that does just that.